packages/datetime/src/dateRangePicker.tsx

```diff
@@ -691,9 +691,10 @@ function getStateChange(value: DateRange,
             } else {
                 if (!leftView.isSame(nextValueStartMonthAndYear)) {
                     leftView = nextValueStartMonthAndYear;
-                }
-                if (!rightView.isSame(nextValueEndMonthAndYear)) {
+                    rightView = nextValueStartMonthAndYear.getNextMonth();
+                } else if (!rightView.isSame(nextValueEndMonthAndYear)) {
                     rightView = nextValueEndMonthAndYear;
+                    leftView = rightView.getPreviousMonth();
```

Comment:
I'd use the same pattern in line 694 and 697 in setting the "other" so it's obvious what's going on